refactor(orc8r): Segregation of Tenants servicer as internal

shivesh-wavelabs opened the pull request:
## Summary
This PR deals with the segregation of tenants.proto internal gRPC servicers implementation.
This changes are part of the ongoing work which details the distinction between Orc8r-internal gRPC endpoints vs. external, gateway-oriented gRPC endpoints.
Moved TenantsService implementation from magma/orc8r/cloud/go/services/tenants/servicers/servicer.go to magma/orc8r/cloud/go/services/tenants/servicers/protected/servicer.go.
Moved tenants.proto implementation from magma/orc8r/protos/tenant.proto to magma/orc8r/cloud/go/tenants/protos/tenant.proto
Verified existing UT cases.

This PR was done as part of discussion by @hcgatewood [RemoveGatewayAccess toOrc8r-InternalEndpoints](https://drive.google.com/file/d/1NO6Qd6rU80xPh0Sb0mKWlSfDt0JbeVIT/view)

## Test Plan

All UTs passed successfully.

shivesh-wavelabs committed:
refactor(orc8r): Segregation of Tenants servicer as internal

Signed-off-by: shivesh <shivesh.ojha@wavelabs.ai>

bbarritt commented:
@hcgatewood a reminder on this PR review request

hcgatewood commented on orc8r/cloud/go/services/tenants/client_api.go:
```diff
@@ -20,8 +20,9 @@ import (
+	"magma/orc8r/cloud/go/services/tenants/protos"
 	merrors "magma/orc8r/lib/go/errors"
-	"magma/orc8r/lib/go/protos"
+	common_protos "magma/orc8r/lib/go/protos"
```
One nit: let's not alias this one (so it's just `protos`), and instead alias the the more-specific protos (so e.g. `tenant_protos`)

hcgatewood reviewed (commented):
Lgtm. Will review in more detail tmrw

christinewang5 commented:
Hmm why the extra `protected` folder in `magma/orc8r/cloud/go/services/tenants/servicers/protected/servicer.go`? All the other services follow the pattern `magma/orc8r/cloud/go/services/**/servicers/servicer.go`?

hcgatewood reviewed (changes requested):
Lgtm, just make that 1 readability change then I'm content 👍

hcgatewood commented:
> Hmm why the extra `protected` folder in `magma/orc8r/cloud/go/services/tenants/servicers/protected/servicer.go`? All the other services follow the pattern `magma/orc8r/cloud/go/services/**/servicers/servicer.go`?

Reasonable question. For context, we're moving all servicers into `**/servicers/{protected,southbound}/` to clearly denote whether the servicer is for internal Orc8r usage vs. southbound exposed to the world. If you wanna read more: https://fb.quip.com/Bb6cA6G9rRIx

shivesh-wavelabs commented:
Hi @hcgatewood, could you review this PR.

hcgatewood commented:
Hi @shivesh-wavelabs, I still am reading a lot of "common_protos" throughout the change. Here's my request to rename those https://github.com/magma/magma/pull/10902#discussion_r793129459

shivesh-wavelabs commented:
> Hi @shivesh-wavelabs, I still am reading a lot of "common_protos" throughout the change. Here's my request to rename those [#10902 (comment)](https://github.com/magma/magma/pull/10902#discussion_r793129459)

@hcgatewood  i have introduced tenants_protos and removed common_protos from all the places.
could you review this PR.
